watchman/decorators.py

```diff
@@ -7,24 +8,50 @@
 
 from watchman import settings
 
+logger = logging.getLogger('watchman')
+
 
 def check(func):
     """
     Decorator which wraps checks and returns an error response on failure.
     """
     def wrapped(*args, **kwargs):
+        check_name = func.__name__
+        arg_name = None
+        if args:
+            arg_name = args[0]
         try:
+            if arg_name:
+                logger.debug("Checking '%s' for '%s'", check_name, arg_name)
+            else:
+                logger.debug("Checking '%s'", check_name)
             response = func(*args, **kwargs)
         except Exception as e:
+            message = str(e)
             response = {
                 "ok": False,
-                "error": str(e),
+                "error": message,
                 "stacktrace": traceback.format_exc(),
             }
             # The check contains several individual checks (e.g., one per
             # database). Preface the results by name.
-            if args:
-                response = {args[0]: response}
+            if arg_name:
+                response = {arg_name: response}
+                logger.exception(
+                    "Error calling '%s' for '%s': %s('%s')",
+                    check_name,
+                    arg_name,
+                    e.__class__.__name__,
+                    message
+                )
+            else:
+                logger.exception(
+                    "Error calling '%s': %s('%s')",
+                    check_name,
+                    e.__class__.__name__,
+                    message
+                )
```

Comment:
@ravangen does this look better to you for the error logging?

Reply:
I personally don't think that the class name in the message adds value here as it is still in the full traceback. I would recommend having the message be targeted for human consumption. If the exception class name is considered valuable, I would suggest putting it in `extra` and let the application's logging formatter determine if it is output.

```python
logger.exception(
    "Error calling '%s': %s('%s')",
    check_name,
    message,
    extra={'exc_class': e.__class__.__name__}
)
```

Reply:
@dhoffman34 Thoughts?

Reply:
@mwarkentin @ravangen, I was thinking that including the class name might make messages clearer by providing more context, as well as helping to distinguish between errors that don't have a message or that have the same message but a different exception class. This could help with services like Sentry that automatically try to group issues.

But based on looking through a bunch of errors, the cases where including the class name would help are pretty unusual. Usually the message alone without the class name is quite clear, and exceptions will almost always have messages that are unique.

Since it doesn't have a high value, I have removed the class name, which will make the messages more readable.